test/Command.js

```diff
@@ -187,7 +184,75 @@ describe("Command", () => {
   });
 
   describe(".runPreparations()", () => {
-    it("needs tests");
+    describe(".filteredPackages", () => {
+      let testDir;
+
+      beforeEach(() => initFixture("UpdatedCommand/basic").then((dir) => {
+        testDir = dir;
+      }));
+
+      class TestCommand extends Command {}
+
+      function cli(cmd, ...args) {
+        return execa.sync(cmd, args, { cwd: testDir }).stdout.toString("utf-8");
+      }
+
+      function run(opts) {
+        const cmd = new TestCommand([], opts, testDir);
+        cmd.run();
+        return cmd;
+      }
+
+      it("--scope should filter packages", () => {
+        const { filteredPackages } = run({ scope: ["package-2", "package-4"] });
+        expect(filteredPackages.length).toEqual(2);
+        expect(filteredPackages[0].name).toEqual("package-2");
+        expect(filteredPackages[1].name).toEqual("package-4");
+      });
+
+      it("--scope-updated should return all packages if no tag is found", () => {
+        const { filteredPackages } = run({ scopeUpdated: "" });
+        expect(filteredPackages.length).toEqual(5);
+      });
+
+      it("--scope-updated should return packages updated since the last tag", () => {
+        cli("git", "tag", "1.0.0");
+        cli("touch", "packages/package-2/random-file");
+        cli("git", "add", ".");
+        cli("git", "commit", "-m", "test");
+
+        const { filteredPackages } = run({ scopeUpdated: "" });
+        expect(filteredPackages.length).toEqual(2);
+        expect(filteredPackages[0].name).toEqual("package-2");
+        expect(filteredPackages[1].name).toEqual("package-3");
```

Comment:
Why is it that `package-3` appears in the list when `package-2` has changed? This is the default behaviour of the `UpdatedPackagesCollector`.

It would appear that packages that have `devDependencies` on packages that have changed get added to the list. This is more of a question than anything else, I guess, so that I understand it better.